service/tbc/peermanager.go

```diff
@@ -436,7 +436,10 @@ func (pm *PeerManager) Run(ctx context.Context) error {
 			go pm.connectSlot(ctx, p)
 
 		case <-ctx.Done():
-			log.Infof("exit")
+			err := ctx.Err()
+			if errors.Is(err, context.Canceled) {
+				return nil
+			}
 			return ctx.Err()
```

Comment:
`err` is defined and only used once in the if statement. Maybe it could be used as the return value as well?
```suggestion
			err := ctx.Err()
			if errors.Is(err, context.Canceled) {
				return nil
			}
			return err
```
or
```suggestion
			if errors.Is(ctx.Err(), context.Canceled) {
				return nil
			}
			return ctx.Err()
```